### PR TITLE
Add Notably UI Skeleton [View]

### LINKED ISSUE #130: Create UI skeleton [View]
1. Create the appropriate controllers and fxml files to support the SidebarTreeview and BlockInputView. 
2. Render the components using hard-coded dummy data.

### PR DESCRIPTION
**Pull request by HemanshuGandhi**
Currently, view components are implemented using existing AB3 architecture. 
Dummy data is used in order to populate components with sample data.

Screen Shot 2020-03-17 at 10 54 49 PM

### CODE REVIEW

File: src/main/java/com/notably/view/BlockContentViewImpl.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package com.notably.view;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.TextArea;
+import javafx.scene.layout.Region;
+
+/**
+ *
+ */
+public class BlockContentViewImpl extends ViewPart<Region> {
```
Comment (kevinputera) [inline src/main/java/com/notably/view/BlockContentViewImpl.java]: Hmmm, we don't really need to call it `BlockContentViewImpl`? We can just call it `BlockContentView` directly

Comment (kevinputera) [reply src/main/java/com/notably/view/BlockContentViewImpl.java]: But, perhaps, you would want to call this `BlockContentEditView` (the text area view)? I think we need to differentiate it from `BlockContentDisplayView` (just a display) somehow. Maybe we can make this change in a future PR?

File: src/main/java/com/notably/view/SideBarTreeViewImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import javafx.fxml.FXML;
+import javafx.scene.control.TreeItem;
+import javafx.scene.control.TreeView;
+import javafx.scene.layout.Region;
+
+/**
+ *
+ */
+public class SideBarTreeViewImpl extends ViewPart<Region> {
```
Comment (kevinputera) [inline src/main/java/com/notably/view/SideBarTreeViewImpl.java]: Same here, we can just call it `SideBarTreeView` I believe

File: src/main/java/com/notably/view/SideBarTreeViewImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        root.setExpanded(true);
+        TreeItem<String> notably = new TreeItem<>("Notably");
+        notably.setExpanded(true);
+        root.getChildren().add(notably);
+        notably.getChildren().add(new TreeItem<>("Another Note"));
+        notably.getChildren().add(new TreeItem<>("CS2103"));
+        notably.getChildren().add(new TreeItem<>("ST2334"));
+
+        sideBarTreeViewRoot.setRoot(root);
+    }
```
Comment (kevinputera) [inline src/main/java/com/notably/view/SideBarTreeViewImpl.java]: Haha interesting that you use `TreeItem` to render the sidebar too!

Currently, @firzanarmani is working on a PR (#125) that implements our block tree using instances of `TreeItem` too. Perhaps we can find a way to make these work together nicely.

File: src/main/java/com/notably/view/MainWindow.java [hunk 4]
```diff
@@ -32,16 +32,24 @@
...
     private MenuItem helpMenuItem;
 
+    @FXML
+    private StackPane sideBarPlaceholder;
+
     @FXML
     private StackPane statusbarPlaceholder;
 
+    @FXML
+    private StackPane textFieldPlaceholder;
```
Comment (kevinputera) [inline src/main/java/com/notably/view/MainWindow.java]: Should we call this `blockContentPlaceholder`? In the future, we can stack either `BlockContentEditView` or `BlockContentDisplayView` on top of this placeholder (pretty sure you thought about this, just reiterating LOL!)

**Review by kevinputera — COMMENTED** [review]
Looks good to me! @AY1920S2-CS2103T-W17-2/developers others please help review this!

File: src/main/java/com/notably/view/BlockContentViewImpl.java (hunk 1 above)
Comment (HemanshuGandhi) [reply src/main/java/com/notably/view/BlockContentViewImpl.java]: For now, I'll change it to BlockContentEditView! If a future PR needs the distinction I will do it then :)

File: src/main/java/com/notably/view/MainWindow.java (hunk 4 above)
Comment (HemanshuGandhi) [reply src/main/java/com/notably/view/MainWindow.java]: Calling it blockContentPlaceholder. Thanks, I understand why this will be useful in the future :)

**Review by johannagwan — COMMENTED** [review]
Hi @HemanshuGandhi, great job!!

Just wanna give a suggestion, perhaps you may also want to explore how to display the heading (which is the same as the name of the Block) as seen in our Mock-up here? It gives more clarity about the user's current working directory, from the user's POV. Thanks!

Screenshot 2020-03-18 at 3 17 11 AM

**Comment by HemanshuGandhi** [comment]
> Hi @HemanshuGandhi, great job!!
> 
> Just wanna give a suggestion, perhaps you may also want to explore how to display the heading (which is the same as the name of the Block) as seen in our Mock-up here? It gives more clarity about the user's current working directory, from the user's POV. Thanks!
> 
> Screenshot 2020-03-18 at 3 17 11 AM

Good idea, I'll open a new issue for this for the timebeing :)

**Review by kevinputera — APPROVED** [review]
Let's merge this in!

**Pull request merged**